docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,230 @@
+Calico Architecture
```

Comment:
Given where we are now (i.e. we're publishing this change as part of the first etcd release) I think this document should be named "architecture.rst".